### ISSUE TITLE
Shortcuts Support

### ISSUE THREAD

**Issue by loca-lhost** [AdguardTeam/AdguardForiOS#1857]
For a future update, can you add support for shortcuts. Thank you!

**Comment by AleksandrShulgin** [comment]
@loca-lhost  Hi there
Thanks for your feature request!
We will consider this.

**Comment by izark99** [comment]
**Please add this feature in next update!!**

**Comment by rakleed** [comment]
@Eugene-Savenko @Versty @artemiv4nov even the macOS app has added support for Shortcuts (https://github.com/AdguardTeam/AdguardForMac/issues/879), although macOS support for Shortcuts appeared later.

Do you plan to add support for Shortcuts in the next update?

For example, I really miss the automation to enable DNS protection after I disconnected from the VPN. Since in Russia all popular VPN services are now actively blocked, and often I have to use the OpenVPN protocol, which disables DNS protection in AdGuard.

**Comment by victorwon** [comment]
It’s almost a year since the initial feature request, even Adguard VPN for iOS has shortcuts now. Could you please speed up the update, or at least give our paid customers this long awaited feature? Thanks

**Comment by artemiv4nov** [comment]
Hi all! 🙌🏼
I understand you, the feature request is important. But we can't inject the functionality to the project as is. Please give us a bit more time to implement the shortcuts functionality.

**Comment by worldsdream** [comment]
It would be nice to have a shortcut where we are able to update the filters.

Right now we have to open the app and refresh the filters lists.

**Comment by RazorII** [comment]
I’d appreciate a simple shortcut to turn adguard on and off.

**Comment by balupton** [comment]
I'm in desperate need of apple shortcuts support for AdGuard.

I have rich automations for unlocking things when certain goals are achieved, and the lack of integration is the Achilles heel of it.

For instance, when I start my day I want to automatically enable my [dns blocklist for enabling agency](https://raw.githubusercontent.com/bevry-labs/filters/master/filter-agency.txt), blocking social/surveillance/addiction media news etc. Then only after I've achieved my goals for the day, which is automatically tracked with apple shortcuts, should social media be unblocked.

ScreenTime is fundamentally broken for this behaviour, as it is too annoying and the annoying modals give you the option to disable it to stop it annoying you, and again there's no good automation for it.

AdGuard blocking is a much better solution.

**Comment by RazorII** [comment]
Here's a concrete use case that makes this more than a nice-to-have:

iOS only lets one VPN tunnel run at a time. AdGuard's DNS protection takes that slot, & so does Tailscale (and WireGuard), so I can't run both & there's no way to automate the swap. Every switch is manual: open AdGuard, turn it off, open Tailscale, turn it on, & reverse it later. A Shortcuts action would make that one tap, or fully automatic (turn DNS protection back on the moment Tailscale disconnects).

What I'd want is small:

1. A "Set DNS protection" action with On / Off / Toggle
2. It returns the resulting status, so it can chain in an automation

Your own apps already ship this: AdGuard VPN for iOS & AdGuard for Mac both have Shortcuts support, so the iOS app is the gap. Paying customer here, & this is the one friction I keep hitting.

Thanks for considering it 🙏

**Comment by hsjeevan** [comment]
Request was originally created in 2021 ? How about move this to high priority ?

**Comment by jhkim0612** [comment]
I have added some extra content to the request. Feel free to read it!
#2552

**Comment by vozersky** [comment]
@jhkim0612 thanks 🙏
